### PR TITLE
gfan 0.7

### PR DESCRIPTION
**Pull request by mahrud**

### CODE REVIEW

File: Formula/gfan.rb [hunk 1]
```diff
@@ -55,31 +49,284 @@ def install
...
++++ b/src/vektor.cpp
+@@ -96,7 +96,7 @@ int gcdGFAN(int r, int s)
+ int gcdOfVector(IntegerVector const &v)
+ {
+   int ret=0;
+-  for(int i=0;i<v.size();i++)if(ret=v[i])break;
++  for(int i=0;i<v.size();i++)if((ret=v[i]))break;
+   if(ret<0)ret=-ret;
+   assert(ret!=0);
+   for(int i=0;i<v.size();i++)ret=gcdGFAN(ret,v[i]);
```
Comment (d-torrance) [inline Formula/gfan.rb:330]: This patch contains the contents of `M2/libraries/gfan/patch-0.6.2`, right?  Do we know why these changes are necessary?  I've only needed the `<cstdint>` patch for the Debian gfan package to build.

Comment (mahrud) [reply Formula/gfan.rb:330]: No, a few things have changed.

Comment (mahrud) [reply Formula/gfan.rb:342]: You're right that the patches seem excessive. I'll look into whether the `exit(1)`s were necessary to fix something in the package.

Comment (mahrud) [reply Formula/gfan.rb:342]: Yeah, basically Dan wanted error signals it seems:
https://github.com/Macaulay2/M2/commit/f088148555806540924ec167f13ea487615c83ee#diff-bd52196e0826f3b0c95eb09f00bc87f1bfedb2c4752f017a631ec6517d28477e

File: Formula/gfan.rb [hunk 2]
```diff
@@ -55,31 +49,332 @@ def install
...
+diff --git a/src/gfanlib_circuittableint.h b/src/gfanlib_circuittableint.h
+index 2b5ced4..609d2d2 100644
+--- a/src/gfanlib_circuittableint.h
++++ b/src/gfanlib_circuittableint.h
+@@ -25,6 +25,7 @@ namespace gfan{
+   template<typename> struct MyMakeUnsigned;
+   template <> struct MyMakeUnsigned<int>{typedef unsigned int type;};
+   template <> struct MyMakeUnsigned<long int>{typedef unsigned long int type;};
++  template <> struct MyMakeUnsigned<long long int>{typedef unsigned long long int type;};
+   template <> struct MyMakeUnsigned<__int128>{typedef unsigned __int128 type;};
```
Comment (mahrud) [inline Formula/gfan.rb:183]: I'm not sure if this makes sense. It seems like support for `__int128` is patchy and on some systems it's just `long long int`, at least that's my guess from errors like
```cpp
src/gfanlib_circuittableint.h:617:57: error: invalid use of incomplete type 'struct gfan::MyMakeUnsigned<long long int>'
  617 |                 typename MyMakeUnsigned<longword>::type positiveResultBoundTimesD=extMul(negabs(s).v,boundA.v);
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~
```
It's a mystery to me why having both declarations makes the compiler happy, but I hope it's not introducing an overflow bug.

File: Formula/gfan.rb (hunk 1 above)
Comment (d-torrance) [reply Formula/gfan.rb:378]: Those `exit`'s aren't in the Debian/Ubuntu, Fedora, or Gentoo gfan packages.

Comment (mahrud) [reply Formula/gfan.rb:378]: Probably the package should be updated to reflect this ...

File: Formula/gfan.rb (hunk 2 above)
Comment (d-torrance) [reply Formula/gfan.rb:183]: These new `__int128`'s are giving me lots of problem in the Debian package, too -- gfan won't even build on 32-bit systems any more.

File: Formula/gfan.rb (hunk 1 above)
Comment (d-torrance) [reply Formula/gfan.rb:378]: I don't understand the point of the `exit`'s.  They all appear in something like this:

```c
assert(0);
exit(1);
```

But `assert(0)` will always abort the program, so we never even reach the call to `exit`.

Comment (mahrud) [reply Formula/gfan.rb:378]: Yeah, I think once the int128 issue is resolved I'll just remove the exits.

**Comment by mahrud** [comment]
@d-torrance I think I figured how to get all these builds to finish at once. It may help with the Ubuntu builds.
